### ISSUE TITLE
Website: TSConfig isolatedDeclarations errors not reported

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#10570]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

From #10001, I'd like to show reports that conflict with [`isolatedDeclarations`](https://www.typescriptlang.org/tsconfig/#isolatedDeclarations). But they're not showing up on the playground.

### Affected URL(s)

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=KYDwDg9gTgLgBAYwgOwM7wGZQgWwMIrpQCuCM0cAvHMsAO5wDKwMAFANoDkAhpwLoBKANwAoEaEixEhTNhwAVAJ5hgAQWTIIMbjACWKAFxMWAHiK7kAcwB8VGvWNsuvQaPHho8JGnhaAFsBQSioAolDYUEbmVnYAjABMAMxCQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0A5gMRUytTt3xwy%2BDtFQZIY6AsjgwAXxBagA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkyJsqAhnugCYAi60zmLhpCmDyYArugA04GpBIjk0PADlRqVAGEAFtwDWZSiPHSAviGNA&tokens=false

### Additional Info

Equivalent TypeScript playground showing that the code does have an error: https://www.typescriptlang.org/play/?isolatedDeclarations=true#code/KYDwDg9gTgLgBAYwgOwM7wGZQgWwMIrpQCuCM0cAvHMsAO5wDKwMAFANoDkAhpwLoBKANwAoEaEixEhTNhwAVAJ5hgAQWTIIMbjACWKAFxMWAHiK7kAcwB8VGvWNsuvQaPHho8JGnhaAFsBQSioAolDYUEbmVnYAjABMAMxCQA

:sparkling

**Comment by ronami** [comment]
Some thoughts while looking into this a bit:

Following up on https://github.com/typescript-eslint/typescript-eslint/issues/10580, it seems that `isolatedDeclarations` isn't working as it's conflicting with `allowJs`, which is enabled [regardless of what's configured on the playground](https://github.com/typescript-eslint/typescript-eslint/blob/4e7f5f86d9b9090fbd9786593802dc6fc94fd134/packages/website/src/components/lib/createCompilerOptions.ts#L15):

```
Option 'allowJs' cannot be specified with option 'isolatedDeclarations'.
```

I think that configuring `allowJs` to be always on (https://github.com/typescript-eslint/typescript-eslint/pull/6785) was necessary so `.js` files work without extra configuration (I assume). Unless this changes on TypeScript's end (https://github.com/microsoft/TypeScript/issues/58262), I think `allowJs` will need to not always be enabled.

As I see it, `allowJs` could be changed to never be enabled automatically, meaning users must enable it manually when choosing a `.js` extension. Alternatively, the playground could set it automatically while handling potential conflicts with options like `isolatedDeclarations`.

Additionally, since `isolatedDeclarations` also errors without having the `declaration` or `composite` options enabled, I think these options will need to either be enabled automatically or have the playground show the relevant errors so developers can adjust the playground's `tsconfig`:

```
Option 'isolatedDeclarations' cannot be specified without specifying option 'declaration' or option 'composite'.
```

**Comment by nayounsang** [comment]
Hello. It seems this issue itself has been resolved. In the example as well, if add the options required for `isolatedDeclarations`, an error is reported. see [this](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=KYDwDg9gTgLgBAYwgOwM7wGZQgWwMIrpQCuCM0cAvHMsAO5wDKwMAFANoDkAhpwLoBKANwAoEaEixEhTNhwAVAJ5hgAQWTIIMbjACWKAFxMWAHiK7kAcwB8VGvWNsuvQaPHho8JGnhaAFsBQSioAolDYUEbmVnYAjABMAMxCQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0A5gMRUytTt3xwy%2BDtFQZIY6AsjgwAXxBagA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkyJsqAhnugCYAi60zmLhpCmDyYArugA04GjASJYJZKyEjxUmlBIjk0PADlRqVAGEAFtwDWZSqvTSAviHtA&tokens=false)

I think `allowJS`(comment from ronami) is a separate issue. Separately from that, I came across a function that overrides `allowJs` in `tsconfig` in `packages/website/src/components/lib/createCompilerOptions.ts`.
